### PR TITLE
Free disk space before setting up python

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #8609

## What changes are proposed in this pull request?

We need to free disk space before setting up python to avid unsetting python.

See this build where:

https://github.com/mlflow/mlflow/actions/runs/5235613256/jobs/9452575980

image

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/actions/free-disk-space/action.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: "free-disk-space"
+description: "free disk space"
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /usr/local/lib/android
```
Comment (harupy) [inline .github/actions/free-disk-space/action.yml:10]: Created this action to avoid repetition.

File: .github/workflows/cross-version-tests.yml [hunk 2]
```diff
@@ -125,11 +126,6 @@ jobs:
         with:
           python-version: ${{ steps.get-python-version.outputs.version }}
       - uses: ./.github/actions/setup-pyenv
-      - name: Remove unneeded libraries
-        run: |
-          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
-          sudo rm -rf /usr/share/dotnet
-          sudo rm -rf /usr/local/lib/android
```
Comment (harupy) [inline .github/workflows/cross-version-tests.yml:132]: Freeing disk space after setting up python unsets the python set up by `actions/setup-pyhton`.

File: mlflow/catboost.py [hunk 3]
```diff
@@ -287,12 +288,10 @@ def _init_model(model_type):
 
     model_types = {c.__name__: c for c in [CatBoost, CatBoostClassifier, CatBoostRegressor]}
 
-    try:
+    with contextlib.suppress(ImportError):
         from catboost import CatBoostRanker
 
         model_types[CatBoostRanker.__name__] = CatBoostRanker
-    except ImportError:
-        pass
```
Comment (harupy) [inline mlflow/catboost.py:295]: A change to run cross-version tests.

**Review by BenWilson2 — APPROVED** [review]
LGTM! Interesting that removing the agent tools wipes the python linkages...

**Pull request merged**